internal/exec/util/passwd.go

```diff
@@ -244,8 +255,19 @@ func (u Util) SetPasswordHash(c types.PasswdUser) error {
 	return err
 }
 
-// CreateGroup creates the group as described.
-func (u Util) CreateGroup(g types.PasswdGroup) error {
+// CreateGroup creates the group as described. If shouldExist variable
+// is set to false and the group already exists then it will be deleted.
+func (u Util) CreateGroup(g types.PasswdGroup, shouldExist bool) error {
```

Comment:
It might make sense to change the function name to `EnsureGroup` as it's no longer aimed at just creating groups but rather ensuring that the state of groups on the system matches the config.